packages/plugin-dlx/sources/commands/dlx.ts

```diff
@@ -43,7 +43,16 @@ export default class DlxCommand extends BaseCommand {
 
       await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`package.json`)), `{}\n`);
       await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`yarn.lock`)), ``);
-      await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`.yarnrc.yml`)), `enableGlobalCache: true\n`);
+
+      const localConfigPath = ppath.join(ppath.cwd(), toFilename(`.yarnrc.yml`));
```

Comment:
`ppath.cwd()` shouldn't be used, because the `cwd` is possible different from the CLI cwd (for example when running `yarn workspaces foreach`, etc). You'll need to call `Configuration.find` and use `configuration.projectCwd`, if available.